contracts/instruments/RibbonThetaVaultYearn.sol

```diff
@@ -0,0 +1,815 @@
+        // solhint-disable-next-line
+        // https://github.com/AlphaFinanceLab/alphahomora/blob/340653c8ac1e9b4f23d5b81e61307bf7d02a26e8/contracts/5/Bank.sol#L104
+        uint256 share =
+            shareSupply == 0 ? amount : amount.mul(shareSupply).div(total);
+
+        require(
+            shareSupply.add(share) >= MINIMUM_SUPPLY,
+            "Insufficient share supply"
+        );
+
+        emit Deposit(msg.sender, amount, share);
+
+        _mint(msg.sender, share);
+    }
+
+    /**
+     * @notice Withdraws ETH from vault using vault shares
+     * @param share is the number of vault shares to be burned
+     */
+    function withdrawETH(uint256 share) external nonReentrant {
+        require(asset == WETH, "!WETH");
+        uint256 withdrawAmount = _withdraw(share, false, true);
+
+        IWETH(WETH).withdraw(withdrawAmount);
+        (bool success, ) = msg.sender.call{value: withdrawAmount}("");
+        require(success, "ETH transfer failed");
+    }
+
+    /**
+     * @notice Withdraws WETH from vault using vault shares
+     * @param share is the number of vault shares to be burned
+     */
+    function withdraw(uint256 share) external nonReentrant {
+        uint256 withdrawAmount = _withdraw(share, false, true);
+        IERC20(asset).safeTransfer(msg.sender, withdrawAmount);
+    }
+
+    /**
+     * @notice Withdraws yvWETH + WETH (if necessary) from vault using vault shares
+     * @param share is the number of vault shares to be burned
+     */
+    function withdrawYieldToken(uint256 share) external nonReentrant {
+        uint256 pricePerYearnShare = collateralToken.pricePerShare();
+        uint256 withdrawAmount =
+            wdiv(_withdraw(share, false, false), pricePerYearnShare);
+
+        (uint256 yieldTokenBalance, uint256 yieldTokensToWithdraw) =
+            _withdrawYieldToken(withdrawAmount, pricePerYearnShare);
+
+        // If there is not enough yvWETH in the vault, it withdraws as much as possible and
+        // transfers the rest in `asset`
+        if (yieldTokensToWithdraw >= yieldTokenBalance) {
+            _withdrawSupplementaryAssetToken(
+                withdrawAmount,
+                yieldTokenBalance,
+                pricePerYearnShare
+            );
+        }
+    }
+
+    /**
+     * @notice Withdraws yvWETH from vault
+     * @param withdrawAmount is the withdraw amount in terms of yearn tokens
+     * @param pricePerYearnShare is the yvWETH<->WETH price ratio
+     */
+    function _withdrawYieldToken(
+        uint256 withdrawAmount,
+        uint256 pricePerYearnShare
+    )
+        private
+        returns (uint256 yieldTokenBalance, uint256 yieldTokensToWithdraw)
+    {
+        yieldTokenBalance = IERC20(address(collateralToken)).balanceOf(
+            address(this)
+        );
+        yieldTokensToWithdraw = max(yieldTokenBalance, withdrawAmount).sub(
+            yieldTokenBalance
+        ) > 0
+            ? yieldTokenBalance
+            : withdrawAmount;
```

Comment:
Can we just simplify this line to

```
yieldTokensToWithdraw = max(yieldTokenBalance, withdrawAmount).sub(yieldTokenBalance);
```

Should be logically equivalent

Reply:
if we do this would have to change logic in other places. 

ex [here](https://github.com/ribbon-finance/structured-products/blob/578933a2c1d3103b848f7548a1844fe5256429b1/contracts/instruments/RibbonThetaVaultYearn.sol#L344) would have to make yieldTokensToWithdraw.sub(yieldtokenbalance) > 0 and then transfer yieldTokensToWithdraw.sub(yieldtokenbalance)

Reply:
Okay, let's keep it this way then